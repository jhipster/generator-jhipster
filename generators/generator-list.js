/**
 * Copyright 2013-2022 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const Generators = {
  GENERATOR_ADD: 'add',
  GENERATOR_APP: 'app',
  GENERATOR_AWS: 'aws',
  GENERATOR_AZURE_APP_SERVICE: 'azure-app-service',
  GENERATOR_AZURE_SPRING_CLOUD: 'azure-spring-cloud',
  GENERATOR_BASE: 'base',
  GENERATOR_BOOTSTRAP: 'bootstrap',
  GENERATOR_BOOTSTRAP_APPLICATION: 'bootstrap-application',
  GENERATOR_CI_CD: 'ci-cd',
  /* @deprecated */
  GENERATOR_CICD: 'ci-cd',
  GENERATOR_CLIENT: 'client',
  GENERATOR_CLOUDFOUNDRY: 'cloudfoundry',
  GENERATOR_COMMON: 'common',
  GENERATOR_CYPRESS: 'cypress',
  GENERATOR_DATABASE_CHANGELOG: 'database-changelog',
  GENERATOR_DATABASE_CHANGELOG_LIQUIBASE: 'database-changelog-liquibase',
  GENERATOR_DOCKER_COMPOSE: 'docker-compose',
  GENERATOR_ENTITIES: 'entities',
  GENERATOR_ENTITIES_CLIENT: 'entities-client',
  GENERATOR_ENTITY: 'entity',
  GENERATOR_ENTITY_CLIENT: 'entity-client',
  GENERATOR_ENTITY_I_18_N: 'entity-i18n',
  /* @deprecated */
  GENERATOR_ENTITY_I18N: 'entity-i18n',
  GENERATOR_ENTITY_SERVER: 'entity-server',
  GENERATOR_EXPORT_JDL: 'export-jdl',
  GENERATOR_GAE: 'gae',
  GENERATOR_GENERATE_BLUEPRINT: 'generate-blueprint',
  GENERATOR_GRADLE: 'gradle',
  GENERATOR_HEROKU: 'heroku',
  GENERATOR_INFO: 'info',
  GENERATOR_INIT: 'init',
  GENERATOR_JAVA: 'java',
  GENERATOR_KUBERNETES: 'kubernetes',
  GENERATOR_KUBERNETES_HELM: 'kubernetes-helm',
  GENERATOR_KUBERNETES_KNATIVE: 'kubernetes-knative',
  GENERATOR_LANGUAGES: 'languages',
  GENERATOR_MAVEN: 'maven',
  GENERATOR_OPENAPI_CLIENT: 'openapi-client',
  GENERATOR_OPENSHIFT: 'openshift',
  GENERATOR_PAGE: 'page',
  GENERATOR_PROJECT_NAME: 'project-name',
  GENERATOR_SERVER: 'server',
  GENERATOR_SPRING_BOOT: 'spring-boot',
  GENERATOR_SPRING_CONTROLLER: 'spring-controller',
  GENERATOR_SPRING_SERVICE: 'spring-service',
  GENERATOR_UPGRADE: 'upgrade',
  GENERATOR_UPGRADE_CONFIG: 'upgrade-config',
  GENERATOR_WORKSPACES: 'workspaces',
};

module.exports = Generators;
