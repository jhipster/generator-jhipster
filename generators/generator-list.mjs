/**
 * Copyright 2013-2023 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project; see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License; Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing; software
 * distributed under the License is distributed on an "AS IS" BASIS;
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND; either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const GENERATOR_ANGULAR = 'angular';
export const GENERATOR_APP = 'app';
export const GENERATOR_AWS = 'aws';
export const GENERATOR_AZURE_APP_SERVICE = 'azure-app-service';
export const GENERATOR_AZURE_SPRING_CLOUD = 'azure-spring-cloud';
export const GENERATOR_BASE = 'base';
export const GENERATOR_BASE_APPLICATION = 'base-application';
export const GENERATOR_BASE_DOCKER = 'base-docker';
export const GENERATOR_BOOTSTRAP = 'bootstrap';
export const GENERATOR_BOOTSTRAP_APPLICATION = 'bootstrap-application';
export const GENERATOR_BOOTSTRAP_APPLICATION_BASE = 'bootstrap-application-base';
export const GENERATOR_BOOTSTRAP_APPLICATION_CLIENT = 'bootstrap-application-client';
export const GENERATOR_BOOTSTRAP_APPLICATION_SERVER = 'bootstrap-application-server';
export const GENERATOR_CASSANDRA = 'cassandra';
export const GENERATOR_CI_CD = 'ci-cd';
export const GENERATOR_CLIENT = 'client';
export const GENERATOR_CLOUDFOUNDRY = 'cloudfoundry';
export const GENERATOR_COMMON = 'common';
export const GENERATOR_COUCHBASE = 'couchbase';
export const GENERATOR_CYPRESS = 'cypress';
export const GENERATOR_DOCKER = 'docker';
export const GENERATOR_DOCKER_COMPOSE = 'docker-compose';
export const GENERATOR_ELASTICSEARCH = 'elasticsearch';
export const GENERATOR_ENTITIES = 'entities';
export const GENERATOR_ENTITY = 'entity';
export const GENERATOR_EXPORT_JDL = 'export-jdl';
export const GENERATOR_GAE = 'gae';
export const GENERATOR_GENERATE_BLUEPRINT = 'generate-blueprint';
export const GENERATOR_GIT = 'git';
export const GENERATOR_GRADLE = 'gradle';
export const GENERATOR_HEROKU = 'heroku';
export const GENERATOR_INFO = 'info';
export const GENERATOR_INIT = 'init';
export const GENERATOR_KAFKA = 'kafka';
export const GENERATOR_KUBERNETES = 'kubernetes';
export const GENERATOR_KUBERNETES_HELM = 'kubernetes-helm';
export const GENERATOR_KUBERNETES_KNATIVE = 'kubernetes-knative';
export const GENERATOR_LANGUAGES = 'languages';
export const GENERATOR_LIQUIBASE = 'liquibase';
export const GENERATOR_LIQUIBASE_CHANGELOGS = 'liquibase-changelogs';
export const GENERATOR_MAVEN = 'maven';
export const GENERATOR_MONGODB = 'mongodb';
export const GENERATOR_OPENAPI_CLIENT = 'openapi-client';
export const GENERATOR_OPENSHIFT = 'openshift';
export const GENERATOR_PAGE = 'page';
export const GENERATOR_PROJECT_NAME = 'project-name';
export const GENERATOR_REACT = 'react';
export const GENERATOR_SERVER = 'server';
export const GENERATOR_SPRING_CONTROLLER = 'spring-controller';
export const GENERATOR_SPRING_SERVICE = 'spring-service';
export const GENERATOR_UPGRADE = 'upgrade';
export const GENERATOR_VUE = 'vue';
export const GENERATOR_WORKSPACES = 'workspaces';
