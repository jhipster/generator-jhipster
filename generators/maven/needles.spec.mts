import { jestExpect as expect } from 'mocha-expect-snapshot';

import { dryRunHelpers as helpers, result as runResult } from '../../test/support/index.mjs';
import BaseApplicationGenerator from '../base-application/index.mjs';
import { GENERATOR_MAVEN } from '../generator-list.mjs';

class mockBlueprintSubGen extends BaseApplicationGenerator {
  constructor(args, opts, features) {
    super(args, opts, features);

    this.sbsBlueprint = true;
  }

  get [BaseApplicationGenerator.POST_WRITING]() {
    return this.asPostWritingTaskGroup({
      mavenStep({ source }) {
        const inProfile = this.options.profile;
        function asItemOrArray<T>(item: T): T | T[] {
          return inProfile ? [item] : item;
        }
        source.addMavenDependencyManagement?.(
          asItemOrArray({
            inProfile,
            groupId: 'dependencyManagementGroupId',
            artifactId: 'dependencyManagementArtifactId',
            version: 'version',
            type: 'type',
            scope: 'scope',
            additionalContent: `
                <exclusions>
                    <exclusion>
                        <groupId>exclusionGroupId</groupId>
                        <artifactId>exclusionArtifactId</artifactId>
                    </exclusion>
                </exclusions>`,
          })
        );

        source.addMavenDistributionManagement?.(
          asItemOrArray({
            inProfile,
            snapshotsId: 'snapshotsId',
            snapshotsUrl: 'snapshotsUrl',
            releasesId: 'releasesId',
            releasesUrl: 'releasesUrl',
          })
        );
        source.addMavenProperty?.(asItemOrArray({ inProfile, property: 'propertyName.dotted', value: 'propertyValue' }));
        source.addMavenDependency?.(
          asItemOrArray({
            inProfile,
            groupId: 'dependencyGroupId',
            artifactId: 'dependencyArtifactId',
            version: 'version',
            additionalContent: `
            <exclusions>
                <exclusion>
                    <groupId>exclusionGroupId</groupId>
                    <artifactId>exclusionArtifactId</artifactId>
                </exclusion>
            </exclusions>`,
          })
        );
        source.addMavenPlugin?.(
          asItemOrArray({
            inProfile,
            groupId: 'mavenPluginGroupId',
            artifactId: 'mavenPluginArtifactId',
            version: 'version',
            additionalContent: `
                <exclusions>
                    <exclusion>
                        <groupId>exclusionGroupId</groupId>
                        <artifactId>exclusionArtifactId</artifactId>
                    </exclusion>
                </exclusions>`,
          })
        );
        source.addMavenPluginManagement?.(
          asItemOrArray({
            inProfile,
            groupId: 'mavenPluginManagementGroupId',
            artifactId: 'mavenPluginManagementArtifactId',
            version: 'version',
            additionalContent: `                    <exclusions>
                        <exclusion>
                            <groupId>exclusionGroupId</groupId>
                            <artifactId>exclusionArtifactId</artifactId>
                        </exclusion>
                    </exclusions>`,
          })
        );
        source.addMavenAnnotationProcessor?.(
          asItemOrArray({
            inProfile,
            groupId: 'annotationProcessorGroupId',
            artifactId: 'annotationProcessorArtifactId',
            version: 'annotationProcessorVersion',
          })
        );
        source.addMavenProfile?.(asItemOrArray({ id: 'profileId', content: '            <other>other</other>' }));
        source.addMavenRepository?.(
          asItemOrArray({
            id: 'repositoryId',
            url: 'repositoryUrl',
            releasesEnabled: true,
            snapshotsEnabled: false,
          })
        );
        source.addMavenPluginRepository?.(
          asItemOrArray({
            id: 'repositoryId',
            url: 'repositoryUrl',
          })
        );
      },
    });
  }
}

describe('generator - maven - needles', () => {
  describe('no profile', () => {
    before(async () => {
      await helpers
        .runJHipster(GENERATOR_MAVEN)
        .withJHipsterConfig({
          blueprint: 'myblueprint',
          clientFramework: 'no',
        })
        .withGenerators([[mockBlueprintSubGen, 'jhipster-myblueprint:maven']]);
    });

    it('Assert pom.xml has the dependency management added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `
            <dependency>
                <groupId>dependencyManagementGroupId</groupId>
                <artifactId>dependencyManagementArtifactId</artifactId>
                <version>version</version>
                <type>type</type>
                <scope>scope</scope>
                <exclusions>
                    <exclusion>
                        <groupId>exclusionGroupId</groupId>
                        <artifactId>exclusionArtifactId</artifactId>
                    </exclusion>
                </exclusions>
            </dependency>`
      );
    });

    it('Assert pom.xml has the distributionManagement added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `    <distributionManagement>
        <snapshotRepository>
            <id>snapshotsId</id>
            <url>snapshotsUrl</url>
        </snapshotRepository>
        <repository>
            <id>releasesId</id>
            <url>releasesUrl</url>
        </repository>
    </distributionManagement>`
      );
    });

    it('Assert pom.xml has the property added', () => {
      runResult.assertFileContent('pom.xml', '<propertyName.dotted>propertyValue</propertyName.dotted>');
    });

    it('Assert pom.xml has the dependency added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `        <dependency>
            <groupId>dependencyGroupId</groupId>
            <artifactId>dependencyArtifactId</artifactId>
            <version>version</version>
            <exclusions>
                <exclusion>
                    <groupId>exclusionGroupId</groupId>
                    <artifactId>exclusionArtifactId</artifactId>
                </exclusion>
            </exclusions>
        </dependency>`
      );
    });
    it('Assert pom.xml has the maven plugin added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `            <plugin>
                <groupId>mavenPluginGroupId</groupId>
                <artifactId>mavenPluginArtifactId</artifactId>
                <version>version</version>
                <exclusions>
                    <exclusion>
                        <groupId>exclusionGroupId</groupId>
                        <artifactId>exclusionArtifactId</artifactId>
                    </exclusion>
                </exclusions>
            </plugin>`
      );
    });

    it('Assert pom.xml has the maven plugin management added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `                <plugin>
                    <groupId>mavenPluginManagementGroupId</groupId>
                    <artifactId>mavenPluginManagementArtifactId</artifactId>
                    <version>version</version>
                    <exclusions>
                        <exclusion>
                            <groupId>exclusionGroupId</groupId>
                            <artifactId>exclusionArtifactId</artifactId>
                        </exclusion>
                    </exclusions>
                </plugin>`
      );
    });

    it('Assert pom.xml has the annotation processor added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `
                            <path>
                                <groupId>annotationProcessorGroupId</groupId>
                                <artifactId>annotationProcessorArtifactId</artifactId>
                                <version>annotationProcessorVersion</version>
                            </path>`
      );
    });

    it('Assert pom.xml has the profile added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `
        <profile>
            <id>profileId</id>
            <other>other</other>
        </profile>`
      );
    });
  });
  describe('prod profile', () => {
    before(async () => {
      await helpers
        .runJHipster(GENERATOR_MAVEN)
        .withJHipsterConfig({
          blueprint: 'myblueprint',
          clientFramework: 'no',
        })
        .withOptions({
          profile: 'prod',
        })
        .withGenerators([[mockBlueprintSubGen, 'jhipster-myblueprint:maven']]);
    });

    it('Assert pom.xml has the dependency management added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `
                    <dependency>
                        <groupId>dependencyManagementGroupId</groupId>
                        <artifactId>dependencyManagementArtifactId</artifactId>
                        <version>version</version>
                        <type>type</type>
                        <scope>scope</scope>
                        <exclusions>
                            <exclusion>
                                <groupId>exclusionGroupId</groupId>
                                <artifactId>exclusionArtifactId</artifactId>
                            </exclusion>
                        </exclusions>
                    </dependency>`
      );
    });

    it('Assert pom.xml has the distributionManagement added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `
            <distributionManagement>
                <snapshotRepository>
                    <id>snapshotsId</id>
                    <url>snapshotsUrl</url>
                </snapshotRepository>
                <repository>
                    <id>releasesId</id>
                    <url>releasesUrl</url>
                </repository>
            </distributionManagement>`
      );
    });

    it('Assert pom.xml has the property added', () => {
      runResult.assertFileContent('pom.xml', '<propertyName.dotted>propertyValue</propertyName.dotted>');
    });

    it('Assert pom.xml has the dependency added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `
                <dependency>
                    <groupId>dependencyGroupId</groupId>
                    <artifactId>dependencyArtifactId</artifactId>
                    <version>version</version>
                    <exclusions>
                        <exclusion>
                            <groupId>exclusionGroupId</groupId>
                            <artifactId>exclusionArtifactId</artifactId>
                        </exclusion>
                    </exclusions>
                </dependency>`
      );
    });
    it('Assert pom.xml has the maven plugin added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `            <plugin>
                        <groupId>mavenPluginGroupId</groupId>
                        <artifactId>mavenPluginArtifactId</artifactId>
                        <version>version</version>
                        <exclusions>
                            <exclusion>
                                <groupId>exclusionGroupId</groupId>
                                <artifactId>exclusionArtifactId</artifactId>
                            </exclusion>
                        </exclusions>
                    </plugin>`
      );
    });

    it('Assert pom.xml has the maven plugin management added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `                <plugin>
                            <groupId>mavenPluginManagementGroupId</groupId>
                            <artifactId>mavenPluginManagementArtifactId</artifactId>
                            <version>version</version>
                            <exclusions>
                                <exclusion>
                                    <groupId>exclusionGroupId</groupId>
                                    <artifactId>exclusionArtifactId</artifactId>
                                </exclusion>
                            </exclusions>
                        </plugin>`
      );
    });

    it('Assert pom.xml has the annotation processor added', () => {
      runResult.assertFileContent(
        'pom.xml',
        `
                                    <path>
                                        <groupId>annotationProcessorGroupId</groupId>
                                        <artifactId>annotationProcessorArtifactId</artifactId>
                                        <version>annotationProcessorVersion</version>
                                    </path>`
      );
    });

    it('should match generated pom', () => {
      expect(runResult.getSnapshot('**/pom.xml')).toMatchSnapshot();
    });
  });
});
