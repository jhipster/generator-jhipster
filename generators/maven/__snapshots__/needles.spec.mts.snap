// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - maven - needles prod profile should match generated pom 1`] = `
{
  "pom.xml": {
    "contents": "<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <repositories>
        <repository>
            <id>repositoryId</id>
            <url>repositoryUrl</url>
            <releases>
                <enabled>true</enabled>
            </releases>
            <snapshots>
                <enabled>false</enabled>
            </snapshots>
        </repository>
    </repositories>

    <pluginRepositories>
        <pluginRepository>
            <id>repositoryId</id>
            <url>repositoryUrl</url>
        </pluginRepository>
    </pluginRepositories>

    <profiles>
        <profile>
            <id>prod</id>
            <dependencyManagement>
                <dependencies>
                    <dependency>
                        <groupId>dependencyManagementGroupId</groupId>
                        <artifactId>dependencyManagementArtifactId</artifactId>
                        <version>version</version>
                        <type>type</type>
                        <scope>scope</scope>
                        <exclusions>
                            <exclusion>
                                <groupId>exclusionGroupId</groupId>
                                <artifactId>exclusionArtifactId</artifactId>
                            </exclusion>
                        </exclusions>
                    </dependency>
                </dependencies>
            </dependencyManagement>
            <distributionManagement>
                <snapshotRepository>
                    <id>snapshotsId</id>
                    <url>snapshotsUrl</url>
                </snapshotRepository>
                <repository>
                    <id>releasesId</id>
                    <url>releasesUrl</url>
                </repository>
            </distributionManagement>
            <properties>
                <propertyName.dotted>propertyValue</propertyName.dotted>
            </properties>
            <dependencies>
                <dependency>
                    <groupId>dependencyGroupId</groupId>
                    <artifactId>dependencyArtifactId</artifactId>
                    <version>version</version>
                    <exclusions>
                        <exclusion>
                            <groupId>exclusionGroupId</groupId>
                            <artifactId>exclusionArtifactId</artifactId>
                        </exclusion>
                    </exclusions>
                </dependency>
            </dependencies>
            <build>
                <plugins>
                    <plugin>
                        <groupId>mavenPluginGroupId</groupId>
                        <artifactId>mavenPluginArtifactId</artifactId>
                        <version>version</version>
                        <exclusions>
                            <exclusion>
                                <groupId>exclusionGroupId</groupId>
                                <artifactId>exclusionArtifactId</artifactId>
                            </exclusion>
                        </exclusions>
                    </plugin>
                </plugins>
                <pluginManagement>
                    <plugins>
                        <plugin>
                            <groupId>mavenPluginManagementGroupId</groupId>
                            <artifactId>mavenPluginManagementArtifactId</artifactId>
                            <version>version</version>
                            <exclusions>
                                <exclusion>
                                    <groupId>exclusionGroupId</groupId>
                                    <artifactId>exclusionArtifactId</artifactId>
                                </exclusion>
                            </exclusions>
                        </plugin>
                        <plugin>
                            <groupId>org.apache.maven.plugins</groupId>
                            <artifactId>maven-compiler-plugin</artifactId>
                            <configuration>
                                <annotationProcessorPaths>
                                    <path>
                                        <groupId>annotationProcessorGroupId</groupId>
                                        <artifactId>annotationProcessorArtifactId</artifactId>
                                        <version>annotationProcessorVersion</version>
                                    </path>
                                </annotationProcessorPaths>
                            </configuration>
                        </plugin>
                    </plugins>
                </pluginManagement>
            </build>
        </profile>
        <profile>
            <id>profileId</id>
            <other>other</other>
        </profile>
    </profiles>
</project>
",
    "stateCleared": "modified",
  },
}
`;
